### PR TITLE
Extract classroom

### PR DESCRIPTION
**Pull request by snicki13**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -40,6 +40,7 @@ services:
       - JITSI_USER=fbs
       - JWT_SECRET=8Dsupersecurekeydf0
       - BBB_SECRET
+      - BBB_URI=http://host.docker.internal:8085
```
Comment (mxsph) [inline docker-compose.yml:43]: For `host.docker.internal` to work under linux the following must be added to the `http` container in `docker-compose.yml`:

```yml
extra_hosts:
      - "host.docker.internal:host-gateway"
```

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      run: ./gradlew dist
+    - name: Build the Docker image
+      run: docker build ws --tag thmmniii/http:classroom
+    - name: Login to Docker Hub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_LOGIN }}
+        password: ${{ secrets.DOCKER_PWD }}
+    - name: Push the Docker image
+      run: docker push thmmniii/http:classroom
```
Comment (mxsph) [inline .github/workflows/docker-image.yml:32]: We should remove this file before merging

**Review by mxsph — COMMENTED** [review]
Already a couple of notes:

- Shouldn't we add the `digital-classroom` to the `docker-compose.yml` (as image with `docker pull ghcr.io/thm-mni-ii/digital-classroom:master`)? So that the FBS can be tested more easily during development.

File: web-gui/src/app/page-components/course-detail/course-detail.component.ts [hunk 3]
```diff
@@ -37,7 +35,8 @@ export class CourseDetailComponent implements OnInit {
   constructor(private taskService: TaskService,
               private authService: AuthService,
               private route: ActivatedRoute, private titlebar: TitlebarService,
-              private conferenceService: ConferenceService, private classroomService: ClassroomService,
+              private conferenceService: ExternalClassroomHandlingService,
```
Comment (mxsph) [inline web-gui/src/app/page-components/course-detail/course-detail.component.ts:38]: This service is no longer used

File: web-gui/src/app/service/external-classroom.service.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+})
+export class ExternalClassroomService {
+  private dialog: MatDialog;
+  private conferenceWindowHandle: Window;
+  private isWindowHandleOpen: Subject<Boolean>;
+  private courseId = 0;
+
+  public constructor(private authService: AuthService,
+                     private classRoomHandlingService: ExternalClassroomHandlingService,
+                     private mDialog: MatDialog,
```
Comment (mxsph) [inline web-gui/src/app/service/external-classroom.service.ts:23]: These three service are no longer used

File: web-gui/src/app/service/external-classroom-handling-service.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  private bbbConferenceLink: BehaviorSubject<object>;
+  private conference: BehaviorSubject<Conference>;
+  public conferenceTimeoutTimer: Subscription;
+
+  public constructor() {
+     this.personalConferenceLink = new BehaviorSubject<string>(null);
+     this.bbbConferenceLink = new  BehaviorSubject<object>(null);
+     this.conference = new BehaviorSubject<Conference>(null);
+  }
+}
```
Comment (mxsph) [inline web-gui/src/app/service/external-classroom-handling-service.ts:23]: This service is no longer used (If you remove this service from the files which only import it, but do not use it)

File: web-gui/src/app/util/rx-stomp.ts [hunk 6]
```diff
@@ -1,80 +1 @@
-import {Client, CompatClient, Frame, IMessage, Stomp, StompSubscription} from '@stomp/stompjs';
-import {Observable, Subscriber} from 'rxjs';
-import * as SockJS from 'sockjs-client';
 import {Subscription} from 'stompjs';
```
Comment (mxsph) [inline web-gui/src/app/util/rx-stomp.ts:1]: @Zitrone44  when this file is removed, several compiler errors occur (see [here](https://github.com/thm-mni-ii/feedbacksystem/runs/4425128466?check_suite_focus=true#step:7:122)) which have nothing to do with it, any idea why?

**Review by mxsph — CHANGES_REQUESTED** [review]
- [ ] When I try to join the classroom I get this error:

```
Could not create JAXBContext for class [class de.thm.ii.fbs.model.classroom.JoinRoomBBBResponse]: Implementation of JAXB-API has not been found on module path or classpath.; nested exception is javax.xml.bind.JAXBException: Implementation of JAXB-API has not been found on module path or classpath. - with linked exception:[java.lang.ClassNotFoundException: com.sun.xml.internal.bind.v2.ContextFactory]
```
- [ ] The conference settings should be removed (since they no longer work anyway).
  
  grafik

- [ ] A user with the global role `Admin` should have the role `DOCENT` in all courses in the Digital Classroom.
- [ ] A user with the Global role 'Moderator' should have the role 'TUTOR' in all courses in the Digital Classroom.

**Comment by snicki13** [comment]
I don't want to know.

File: web-gui/src/app/util/rx-stomp.ts (hunk 6 above)
Comment (mxsph) [reply web-gui/src/app/util/rx-stomp.ts]: I have found the error/a solution (see ee429f7642cd82a3ce1156d9554dac08ea94db9f)

**Review by mxsph — CHANGES_REQUESTED** [review]
I have found another bug where it is possible to end up in two different classrooms in the same course:

User 1: Create & Join Course
User 2: Join Course
User 1: Open Classroom
User 1: Close Classroom
restart digital-classroom (`docker-compose restart digital-classroom`)
User 1: Open Classroom
User 2: Open Classroom

Now the two users are in two different classrooms

**Comment by snicki13** [comment]
> I have found another bug where it is possible to end up in two different classrooms in the same course:
> 
> User 1: Create & Join Course User 2: Join Course User 1: Open Classroom User 1: Close Classroom restart digital-classroom (`docker-compose restart digital-classroom`) User 1: Open Classroom User 2: Open Classroom
> 
> Now the two users are in two different classrooms

good catch

**Review by mxsph — APPROVED** [review]

**Pull request merged**